invenio_rdm_migrator/transform/transactions.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio RDM migration transaction group transform."""
+
+from .base import Transform
+
+
+class TransactionGroup(Transform):
+    """Transform a transaction group and its items."""
+
+    def __init__(self, table_entry_map):
+        """Constructor.
+
+        :param table_entry_map: a dictionary mapping entry transformers to table names.
+        """
+        # QUESTION: should this dict be a separate class including some validation
+        self.table_entry_map = table_entry_map
+
+    def _transform_data(self, table, data):
+        """Transform a specific table data based on the defined entry."""
+        entry = self.table_entry_map.get("table")
+
+        return entry.transform(data) if entry else data
+
+    def _transform(self, entry):
+        """Transform entry.
+
+        An entry is a dictionary with a list of operations.
+        """
+        operations = entry["operations"]
+        # this information refers to the semantic meaning of the group
+        # for example: record metadata update, file upload, draft creation, etc.
+        tx_group_id = entry["tx_group_id"]
+        # the actual transaction id, which would be useful for debug and error handling
+        tx_id = entry["tx_id"]
+
+        for operation in operations:
+            payload = operation["payload"]
+            table_name = payload["source"]["table"]
+            yield {
```

Comment:
self note: it does not make much sense to do this split, it defeats the purpose of the previous grouping. It should be passed as a whole to the `Load`; maybe even be commited as 1 in the session.